### PR TITLE
Miscellaneous code tidies in EventSchedule and other components

### PR DESCRIPTION
**Pull request by alexwlchan**
## Who is this for?

Devs.

## What is it doing for them?

Tightening up some of this code; reusing components for consistency. For https://github.com/wellcomecollection/wellcomecollection.org/issues/7831

### CODE REVIEW

File: content/webapp/components/FeaturedCard/FeaturedCard.tsx [hunk 1]
```diff
@@ -149,13 +149,9 @@ const FeaturedCardExhibitionBody = ({
           className={`${font('intr', 4)} no-margin no-padding`}
         >
           <>
-            <time dateTime={exhibition.start.toUTCString()}>
```
Comment (rcantin-w) [inline content/webapp/components/FeaturedCard/FeaturedCard.tsx:152]: Was there a reason for this one to be UTC? But I'm thinking this is also to align based on the docs that Danny shared on Slack

Comment (alexwlchan) [reply content/webapp/components/FeaturedCard/FeaturedCard.tsx:152]: I think this was a mistake – I can't think of why the start would be UTC and the end would be ISO.

**Review by rcantin-w — APPROVED** [review]

File: content/webapp/pages/whats-on.tsx [hunk 2]
```diff
@@ -159,9 +160,7 @@ const DateRange = ({ dateRange, period }: DateRangeProps) => {
       as="p"
       className={font('intr', 5)}
     >
-      {period === 'today' && (
-        <time dateTime={formatDate(start)}>{formatDate(start)}</time>
-      )}
+      {period === 'today' && <HTMLDate date={start} />}
       {period === 'this-weekend' && (
         <>
           <time dateTime={formatDate(start)}>{formatDay(start)}</time>
```
[the remark this answers is not in the document]
Comment (rcantin-w) [reply content/webapp/pages/whats-on.tsx:166]: To be fair they're similar looking words and I probs looked too quickly, so defo not a blocker :)

[the remark this answers is not in the document]
Comment (alexwlchan) [reply content/webapp/pages/whats-on.tsx]: I've renamed the method to `formatDayName`, because I've got confused by this as well! :)

**Review by rcantin-w — APPROVED** [review]
Great!

**Pull request merged**